### PR TITLE
Create 0028-find-the-index-of-the-first-occurrence-in-a-string.js

### PR DESCRIPTION
**Pull request by aadil42**
Added solution for find-the-index-of-the-first-occurrence-in-a-string in JS

File(s) Added: 0028-find-the-index-of-the-first-occurrence-in-a-string.js
Language(s) Used: JavaScript
Submission URL: https://leetcode.com/problems/find-the-index-of-the-first-occurrence-in-a-string/submissions/985281830/

### CODE REVIEW

File: javascript/0028-find-the-index-of-the-first-occurrence-in-a-string.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            } else if(j !== 0 && needle[j] === needle[i]) {
+                pattern[i] = 1 + pattern[j - 1];
+                i++;
+            } else {
+                j = pattern[j - 1];
+            }
+        }
+    }
+    
+j = 0;
```
Comment (aakhtar3) [inline javascript/0028-find-the-index-of-the-first-occurrence-in-a-string.js:34]: Suggestion: fix empty space and be consistent

File: javascript/0028-find-the-index-of-the-first-occurrence-in-a-string.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            i++;
+            j++;
+        } else if(needle[i] !== needle[j]) {
+            if(j === 0) {
+                pattern[i] = 0;
+                i++;
+            } else if(j !== 0 && needle[j] === needle[i]) {
+                pattern[i] = 1 + pattern[j - 1];
+                i++;
+            } else {
```
Comment (aakhtar3) [inline javascript/0028-find-the-index-of-the-first-occurrence-in-a-string.js:28]: Suggestion: simplify your conditions and avoid else

**Review by aakhtar3 — CHANGES_REQUESTED** [review]

File: javascript/0028-find-the-index-of-the-first-occurrence-in-a-string.js (hunk 2 above)
Comment (aadil42) [reply javascript/0028-find-the-index-of-the-first-occurrence-in-a-string.js]: done.

File: javascript/0028-find-the-index-of-the-first-occurrence-in-a-string.js (hunk 1 above)
Comment (aadil42) [reply javascript/0028-find-the-index-of-the-first-occurrence-in-a-string.js]: the spacing is fixed.

File: javascript/0028-find-the-index-of-the-first-occurrence-in-a-string.js (hunk 2 above)
Comment (aadil42) [reply javascript/0028-find-the-index-of-the-first-occurrence-in-a-string.js]: Just a kind reminder that I updated the code as suggested.

**Comment by aadil42** [comment]
The solution to this problem has been added by someone. I'm closing the PR.

**Pull request closed without merging**